xarray/core/dataarray.py

```diff
@@ -1086,6 +1086,35 @@ def unstack(self, dim):
         ds = self._to_temp_dataset().unstack(dim)
         return self._from_temp_dataset(ds)
 
+    def to_unstacked_dataset(self, dim, level=0):
+        """Unstack DataArray expanding to Dataset along a given level
+
+        Parameters
+        ----------
+        dim : str
+            Name of existing dimension to unstack
+        level : int
+            Index of level to expand to dataset along
+
+        Returns
+        -------
+        unstacked: Dataset
+
+        """
+
+        idx = self.indexes[dim]
+        if not isinstance(idx, pd.MultiIndex):
+            raise ValueError(f"{dim} is not a stacked coordinate")
+        variables = idx.levels[level]
+
+        # pull variables out of datarray
+        data_dict = {}
+        for k in variables:
+            data_dict[k] = self.sel(variable=k).squeeze(drop=True)
```

Comment:
should we add a keyword argument `variable_dim` to match the `to_stacked_array` argument?
```Python
data_dict[k] = self.sel(**{variable_dim: k).squeeze(drop=True)
```

Reply:
yes, definitely. good catch.